.github/workflows/nightly_run.yaml

```diff
@@ -0,0 +1,76 @@
+name: Test Update smoke 
+"on":
+  # Going to run the scripts/test_update_smoke.sh once each night at midnight
+  # Currently staying with Postgres 17
+  schedule:
+    - cron: '0 0 * * *'
+  workflow_dispatch:
+  pull_request:
+
+jobs:
+  update_test:
+    # We'll run the test against an instance in the cloud
+    # The previous version of TimescaleDB is installed
+    # and then we upgrade to the current version
+    name: Cloud Update test smoke 
+    runs-on: 'ubuntu-latest'
+    steps:
+    - name: Checkout TimescaleDB
+      uses: actions/checkout@v4
+
+    - name: Install Dependencies
+      # we want the right version of Postgres for handling any dump file
+      run: |
+        sudo apt-get update
+        sudo apt-get install gnupg postgresql-common 
+        yes | sudo /usr/share/postgresql-common/pgdg/apt.postgresql.org.sh
+        sudo apt-get update
+        sudo apt-get install postgresql-17 
+
+    - name: Read versions
+      # Get two of the version parameters for test_update_smoke.sh
+      id: versions
+      run: |
+        # Read current version of TimescaleDB from version.config
+        # version will only be a proper version in a release branch 
+        # so we use previous_version as fallback for main
+        if grep '^version = [0-9.]\+$' version.config; then
+          version=$(grep '^version = ' version.config | sed -e 's!^version = !!')
+        else
+          version=$(grep '^previous_version = ' version.config | sed -e 's!^previous_version = !!')
+        fi
+        echo "version=${version}" >>$GITHUB_OUTPUT
+        # Now get the previous version
+        prev_version=$(wget --quiet -O - \
+        https://raw.githubusercontent.com/timescale/timescaledb/${version}/version.config \
+          | grep previous_version | sed -e 's!previous_version = !!')
+        echo "prev_version=${prev_version}" >>$GITHUB_OUTPUT
+        # And report what we found
+        echo "version=${version}"  "prev_version=${prev_version}"
+
+    - name: Update smoke 
+      # Now run the test.  Currently the cloud instance is always up.
+      # We might want to have a more flexible approach
+      run: |
+        PATH="/usr/lib/postgresql/17/bin:$PATH"
```

Comment:
Nit: You can use the `env` tag: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsenv